app/models/concerns/smooch_messages.rb

```diff
@@ -322,16 +322,21 @@ def save_message_later(message, app_id, request_type = 'default_requests', annot
       queue = RequestStore.store[:smooch_bot_queue].to_s
       queue = queue.blank? ? 'smooch_priority' : (mapping[queue] || 'smooch_priority')
       type = (message['type'] == 'text' && !message['text'][/https?:\/\/[^\s]+/, 0].blank?) ? 'link' : message['type']
-      SmoochWorker.set(queue: queue).perform_in(1.second + interval.seconds, message.to_json, type, app_id, request_type, YAML.dump(annotated))
+      associated_id = annotated&.id
+      associated_class = annotated.class.name
+      SmoochWorker.set(queue: queue).perform_in(1.second + interval.seconds, message.to_json, type, app_id, request_type, associated_id, associated_class)
     end
 
     def default_archived_flag
       team_id = self.config['team_id'].to_i
       Bot::Alegre.team_has_alegre_bot_installed?(team_id) ? CheckArchivedFlags::FlagCodes::PENDING_SIMILARITY_ANALYSIS : CheckArchivedFlags::FlagCodes::NONE
     end
 
-    def save_message(message_json, app_id, author = nil, request_type = 'default_requests', associated_obj = nil)
+    def save_message(message_json, app_id, author = nil, request_type = 'default_requests', associated_id = nil, associated_class = nil)
       message = JSON.parse(message_json)
+      return if TiplineRequest.where(smooch_message_id: message['_id']).last.present?
+      associated_obj = nil
+      associated_obj = begin associated_class.constantize.where(id: associated_id).last rescue nil end unless associated_id.nil?
```

Comment:
Why we need a `begin`...`rescue` here? In which cases we expect it to raise an exception that is fine to ignore?

Reply:
removed `begin` ... `rescue`